internal/provider/deployment.go

```diff
@@ -125,7 +125,7 @@ func (d *DeploymentDeployer) assignNodesIPs() error {
 		}
 		ip := cidr.IP
 		ip[3] = cur
-		for isInByte(usedIPs, ip[3]) {
+		for includes(usedIPs, ip[3]) {
 			if cur == 254 {
 				return errors.New("all 253 ips of the network are exhausted")
```

Comment:
add the network information to the error message, so the user would be able to tell the network and the subnet used